pybedlite/overlap_detector.py

```diff
@@ -126,36 +177,94 @@ def from_bedrecord(cls: Type["Interval"], record: BedRecord) -> "Interval":
         )
 
 
-class OverlapDetector(Iterable[Interval]):
+_GenericGenomicSpan = TypeVar(
+    "_GenericGenomicSpan",
+    bound=Union[_GenomicSpanWithChrom, _GenomicSpanWithContig, _GenomicSpanWithRefName],
+)
+"""
+A generic genomic feature where the reference sequence name is accessed with any of the 3 most
+common property names ("chrom", "contig", "refname"). This type variable is used for describing the
+generic type contained within the :class:`~pybedlite.overlap_detector.OverlapDetector`.
+"""
```

Comment:
This redefinition of `GenomicSpan` is necessary to decouple the generic type parameter of the overlap detector with the types it will receive in any of it's overlap methods.